### PR TITLE
Bugfix/autostart

### PR DESCRIPTION
**Pull request by camilasan**

### CODE REVIEW

File: src/common/utility_mac.mm [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    Q_UNUSED(folder)
+}
+
+static Result<void, QString> writePlistToFile(NSString *plistFile, NSDictionary *plist)
+{
+    NSError *error = nil;
+
+    // Check if the directory exists. On a fresh installation for example, the directory does not
+    // exist, so writing the plist file below will fail.
+    QDir plistDir = QFileInfo(QString::fromNSString(plistFile)).dir();
```
Comment (claucambra) [inline src/common/utility_mac.mm:69]:
```suggestion
    const auto plistDir = QFileInfo(QString::fromNSString(plistFile)).dir();
```

**Review by claucambra — COMMENTED** [review]
Looks good, only one insignificant nitpick

File: src/common/utility_unix.cpp [hunk 2]
```diff
@@ -104,10 +112,12 @@ void setLaunchOnStartup_private(const QString &appName, const QString &guiName,
     }
 }
 
-static inline bool hasDarkSystray_private()
+#ifndef TOKEN_AUTH_ONLY
```
Comment (mgallien) [inline src/common/utility_unix.cpp:115]: I think you should remove this
if `TOKEN_AUTH_ONLY` is defined, the build would most probably fail ?

File: src/common/utility_win.cpp [hunk 3]
```diff
@@ -147,8 +148,8 @@ void setLaunchOnStartup_private(const QString &appName, const QString &guiName,
     }
 }
 
-// TODO: Right now only detection on toggle/startup, not when windows theme is switched while nextcloud is running
-static inline bool hasDarkSystray_private()
+#ifndef TOKEN_AUTH_ONLY
```
Comment (mgallien) [inline src/common/utility_win.cpp:151]: I think you should remove this
if `TOKEN_AUTH_ONLY` is defined, the build would most probably fail ?

File: src/common/utility_mac.mm [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+            // plist doens't exist, write a new one.
+            auto result = writeNewPlistFile(plistFile, fullPath, enable);
+            if (!result) {
+                qCWarning(lcUtility) << result.error();
+            }
+        }
+    }
+}
+
+#ifndef TOKEN_AUTH_ONLY
```
Comment (mgallien) [inline src/common/utility_mac.mm:243]: I think you should remove this
if `TOKEN_AUTH_ONLY` is defined, the build would most probably fail ?

File: src/gui/generalsettings.cpp [hunk 5]
```diff
@@ -456,7 +453,13 @@ void GeneralSettings::saveMiscSettings()
 
 void GeneralSettings::slotToggleLaunchOnStartup(bool enable)
 {
-    Theme *theme = Theme::instance();
+    ConfigFile configFile;
+    if (enable == Utility::hasLaunchOnStartup(Theme::instance()->appName())) {
+        return;
+    }
+
+    const auto theme = Theme::instance();
```
Comment (mgallien) [inline src/gui/generalsettings.cpp:461]: why not move the declaration of `theme` before the `if` line ?
you would then be able to also use it in the `if`

**Review by mgallien — CHANGES_REQUESTED** [review]
build fail if we define `TOKEN_AUTH_ONLY` cmake option to `ON`

**Comment by mgallien** [comment]
> build fail if we define `TOKEN_AUTH_ONLY` cmake option to `ON`

that said, `master` branch compilation is also broken if we define `TOKEN_AUTH_ONLY` to `ON` when configuring the build with `cmake`
meaning, I would not use this macro in modified/new code given the brokenness

**Review by mgallien — APPROVED** [review]

**Pull request merged**